### PR TITLE
Feature/set target framework to netstandard 20

### PR DESCRIPTION
**Pull request by NullableInt**

### CODE REVIEW

File: src/SwedbankPay.Sdk.Infrastructure/Extensions/EnumExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SwedbankPay.Sdk.Extensions
+{
+    public static class EnumExtensions
+    {
+        public static TEnum Parse<TEnum>(String value) where TEnum : struct
```
Comment (asbjornu) [inline src/SwedbankPay.Sdk.Infrastructure/Extensions/EnumExtensions.cs:9]: Make this into a proper extension method so the word `Extensions` make sense.

```suggestion
    public static class StringExtensions
    {
        public static TEnum ToEnum<TEnum>(this string value) where TEnum : struct
```

**Review by asbjornu — CHANGES_REQUESTED** [review]
If we're building for several different framework versions, we need to introduce `matrix` into the GitHub Actions workflow, no? Also, the tests are failing.

**Comment by NullableInt** [comment]
> 
> 
> If we're building for several different framework versions, we need to introduce `matrix` into the GitHub Actions workflow, no? Also, the tests are failing.

Where do you want matrix testing? And on what?

Test failed due to `{"name":"Transactions[1].Activities","description":"Activity chain is broken."}` meaning we have to re run the test after a few minutes. This is not an error in production and only happens on external integration.

File: src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Card/CardPayment.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public CardPayment(CardPaymentDto payment)
             Currency = new Currency(payment.Currency);
             Description = payment.Description;
             Id = new Uri(payment.Id, UriKind.RelativeOrAbsolute);
-            Instrument = EnumExtensions.Parse<PaymentInstrument>(payment.Instrument);
+            Instrument = StringExtensions.Parse<PaymentInstrument>(payment.Instrument);
```
Comment (asbjornu) [inline src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Card/CardPayment.cs:25]:
```suggestion
            Instrument = payment.Instrument.ParseTo<PaymentInstrument>();

```

File: src/SwedbankPay.Sdk.Infrastructure/Extensions/StringExtensions.cs [hunk 3]
```diff
@@ -4,9 +4,10 @@
 
 namespace SwedbankPay.Sdk.Extensions
 {
-    public static class EnumExtensions
+    public static class StringExtensions
+
     {
-        public static TEnum Parse<TEnum>(String value) where TEnum : struct
+        public static TEnum Parse<TEnum>(this string value) where TEnum : struct
```
Comment (asbjornu) [inline src/SwedbankPay.Sdk.Infrastructure/Extensions/StringExtensions.cs:10]:
```suggestion
        public static TEnum ParseTo<TEnum>(this string value) where TEnum : struct
```

File: src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Card/CardPayment.cs [hunk 4]
```diff
@@ -43,7 +43,7 @@ public CardPayment(CardPaymentDto payment)
             Transactions = payment.Transactions?.Map();
             if (!string.IsNullOrEmpty(payment.Intent))
             {
-                Intent = EnumExtensions.Parse<PaymentIntent>(payment.Intent);
+                Intent = StringExtensions.Parse<PaymentIntent>(payment.Intent);
```
Comment (asbjornu) [inline src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Card/CardPayment.cs:46]:
```suggestion
                Intent = payment.Intent.ParseTo<PaymentIntent>();

```

**Review by asbjornu — CHANGES_REQUESTED** [review]
>Where do you want matrix testing? And on what?

[`matrix`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) is used to parameterize every `job` for a different global variable, usually specifying the environment.

We should perform `dotnet build`, `dotnet test`, etc., for each version of .NET we claim to support, no?

**Comment by NullableInt** [comment]
> 
> 
> > Where do you want matrix testing? And on what?
> 
> [`matrix`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) is used to parameterize every `job` for a different global variable, usually specifying the environment.
> 
> We should perform `dotnet build`, `dotnet test`, etc., for each version of .NET we claim to support, no?

I can set this up for the test step. But each run will take approximately 19 minutes.

**Comment by NullableInt** [comment]
> 
> 
> > > Where do you want matrix testing? And on what?
> > 
> > 
> > [`matrix`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) is used to parameterize every `job` for a different global variable, usually specifying the environment.
> > We should perform `dotnet build`, `dotnet test`, etc., for each version of .NET we claim to support, no?
> 
> I can set this up for the test step. But each run will take approximately 19 minutes for each configuration.

Looking into it this will not work as the test projects does not use the nuget (nor should they), so I'll need to come up with a different way of testing this.

**Comment by NullableInt** [comment]
@asbjornu Can we do matrix testing in a different pull request?

**Comment by asbjornu** [comment]
> @asbjornu Can we do matrix testing in a different pull request?

Yes, absolutely.

**Review by asbjornu — APPROVED** [review]

**Pull request merged**